src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,138 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (isEmpty()) {
+            head = tail = newNode;
+        }
+        tail.next = newNode;
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> findCurrentNode = getNodeByIndex(index);
+        Node<T> newNode = new Node<>(findCurrentNode.prev, value, findCurrentNode);
+        if (index == 0) {
+            head = newNode;
+        } else {
+            findCurrentNode.prev.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = getNodeByIndex(index);
```

Comment:
This is repeating code. You always call `checkIndex `before `getNodeByIndex`. Call it inside `getNodeByIndex`
![image](https://user-images.githubusercontent.com/20326858/221957458-2898b4d6-ac9a-41a7-86a3-67f9b163f6cc.png)
![image](https://user-images.githubusercontent.com/20326858/221957516-f27b2b66-370f-478a-b098-85d4b97b47e9.png)